### PR TITLE
MWPW-161897: tagging support

### PR DESCRIPTION
**Pull request by yesil**
Introduces `aem-tag-picker-field` custom element.
Adds tags field in the merch card editor
Updates the filter panel to filter based on tags. 

Resolve: MWPW-161897

Test URLs:
- Before: https://main--mas--adobecom.hlx.live/studio.html
- After: https://mwpw-161897--mas--adobecom.hlx.live/studio.html

### CODE REVIEW

File: studio/src/aem/aem.js
Comment (npeltier) [inline studio/src/aem/aem.js:201]: is newFragment here for debugging?

File: studio/src/aem/aem-tag-picker-field.js
Comment (npeltier) [inline studio/src/aem/aem-tag-picker-field.js:234]: do you think we can add `protected-namespaces` property that contains all subname spaces for which we should _not_ be able to remove tags from UI?

File: studio/src/aem/aem-tag-picker-field.js
Comment (npeltier) [inline studio/src/aem/aem-tag-picker-field.js:105]: await?

**Review by npeltier — APPROVED** [review]

File: package.json
Comment (yesil) [reply package.json:52]: oups, it wasn't intentional, but I guess we don't need tabs just yet.

File: studio/src/aem/aem-tag-picker-field.js
Comment (yesil) [reply studio/src/aem/aem-tag-picker-field.js:234]: for a tag picker, we can restrict the exact tree, the user can add / remove tags from.
And here, only those tags will be shown.
To me it makes sense that the user can remove tags that he might have added.
We can have multiple tag pickers storing tags in the same `tags` field.

File: studio/src/aem/aem-tag-picker-field.js
Comment (yesil) [reply studio/src/aem/aem-tag-picker-field.js:105]: connectedCallback is not async, cannot do await but, I don't need to await neither here.

**Review by 3ch023 — APPROVED** [review]

**Comment by afmicka** [comment]
@yesil Few observations. Let me know what is intended behaviour:
1. Filter pop-up does not close once the selection is made (the last hierarchy value) 
2. Filters are not preserved on page refresh (they are lost if page is reloaded)
3. If you perform combination of search and filter, it does not give correct result in case there is no a match and takes only the value from the search field. (Here i have searched an id card and filtered by creative all apps. This card does not have that tag, i would expect empty list with this combination) 
Screenshot 2024-11-19 at 15 03 40

**Comment by yesil** [comment]
@afmicka this is expected, both contains links to **https://www.adobe.com/creativecloud/plans.html**

image

**Comment by afmicka** [comment]
@yesil on smaller screens (Macbook 13"), if you scroll the edit form almost to the bottom and open the tag menu, the content gets out of the popup. 
Screenshot 2024-11-20 at 10 10 09

**Comment by yesil** [comment]
@afmicka the item overflow on small screen is fixed, checking CTAs.

**Comment by afmicka** [comment]
@yesil @3ch023 @npeltier @Roycethan @Axelcureno 
I keep seeing the same issues with CTA and price edits mentioned above. Given that no relation to the code changes could be identified and the PR is blocking further work, have created separate jira for tracking if this start happening on main as well after merging this. https://jira.corp.adobe.com/browse/MWPW-162870 

If this is ok with everyone, please feel free to merge this PR to unblock.

**Pull request merged**
